### DISCUSSION TITLE
Motivating long-slog projects

### DISCUSSION THREAD

**Discussion by pp-mo** [SciTools/iris#5612] [General]
We have a bit of a history of coming up with ideas that would make widespread changes to the codebase, but then find it hard to follow through to completion.

My personal list of examples ...
  * all tests to be categorised into integration / unit tests
  * ~all public functions to document whether they lazy preserve lazy data or not~
    * an honourable mention, as this https://github.com/SciTools/iris/pull/3293 **_did_** get completely finished, by @ESadek-MO (#5137)
  * numpydoc style for all docstrings
    * ongoing -- see https://github.com/SciTools/iris/issues/4721
  * [pytest adoption ](https://github.com/SciTools/iris/issues/5574)
  * [type hinting](https://github.com/SciTools/iris/discussions/4272)
  * 100% code coverage, by unit tests, of every code module / facility

I'd class these are "code refresh projects" -- they seem to follow a pattern where any/all code may need adjusting to conform with a new desired standard

However, the strategic point here is not the content of such a list, but 
**_HOW_ we can more effectively drive these types of "code refresh projects" to completion**

**Comment by pp-mo** [comment]
Emerged from offline discussion between @pp-mo @trexfeathers @bjlittle 
Some observations + ideas ...
  * problem of ownership
  * the approach https://github.com/SciTools/iris/issues/4721 is useful, but the long list is rather daunting -- probably does not encourage people to join in !
  * if each "work section" is an issue it could seem overwhelming ..
    * .. likewise if they are all in a project
    * however, we could open only a few sections of work as "live" issues at any one time, starting others as they get resolved
  * @bjlittle relates progress towards [full ruff compliance in geovista](https://github.com/bjlittle/geovista/pull/531)
    * divide + conquer by definite sections (a subset of rules)
    * applying iteratively : a measure (#remaining errors) looks manageable at each step
  * we need a goal to be "completion"
  * some kind of numeric measure, like a "%ge compliant" or "remaining issues countdown", would act as a motivator
  *  we could use a sprint effort to divide + conquer with a focussed approach, which can deliver ...
    * team working
    * shared goals
    * recognition of contribution
    * friendly competition

**Comment by trexfeathers** [comment]
Thanks very much for capturing this, @pp-mo!
